core/lib/dal/src/blocks_dal.rs

```diff
@@ -2241,6 +2289,31 @@ impl BlocksDal<'_, '_> {
             .context("storage contains neither L2 blocks, nor snapshot recovery info")?;
         Ok(snapshot_recovery.protocol_version)
     }
+
+    pub async fn set_tree_writes(
+        &mut self,
+        l1_batch_number: L1BatchNumber,
+        tree_writes: Vec<TreeWrite>,
+    ) -> DalResult<()> {
```

Comment:
Dumb question: Why two ways to insert tree writes for an L1 batch? I'd try to retain this one and use it atomically with inserting a batch header if necessary.

Reply:
`set_tree_writes` was added at the last moment for the test. I agree, it would look better to use `set_tree_writes` everywhere, I will change code